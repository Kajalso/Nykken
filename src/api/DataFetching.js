import React, { useState, useEffect } from "react";
import axios from "axios";

import "./api.css";

function DataFetching() {
  const [id, setId] = useState(1);
  const [dataType, setDataType] = useState(1);

  const [loading, setLoading] = useState(true);

  const [measurementsData, setMeasurementsData] = useState([]);
  const [measurements, setMeasurements] = useState([]);

  const [timestamps, setTimestamps] = useState({});

  const [fromTime, setFromTime] = useState("00:00:00");
  const [untilTime, setUntilTime] = useState("00:11:00");

  const [fromDate, setFromDate] = useState("2020-08-08");
  const [untilDate, setUntilDate] = useState("2020-08-08");

  const handleClick = () => {
    setDataType(id);
    console.log("Showing data with ID " + id);
    console.log("From: " + fromDate + " " + fromTime);
    console.log("Until: " + untilDate + " " + untilTime);
    console.log(measurementsData);
    console.log(timestamps);
    console.log(measurements);
  };

  useEffect(() => {
    setLoading(true);
    console.log("Data fetching...");
    axios
      .get(
        `http://ibmrisvol.ibm.ntnu.no/data?from=${fromDate}T${fromTime.slice(
          0,
          2
        )}%3A${fromTime.slice(3, 5)}%3A${fromTime.slice(
          6,
          8
        )}Z&until=${untilDate}T${untilTime.slice(0, 2)}%3A${untilTime.slice(
          3,
          5
        )}%3A${untilTime.slice(6, 8)}Z&identifier=${dataType}`
      )
      .then((response) => {
        setMeasurementsData(response.data.data[0].measurements);
        setMeasurements(
          measurementsData.map((data) => Number(data.measurement))
        );
        setTimestamps(
          measurementsData.map((data) => {
            let date = new Date(data.time_stamp_utc);
            return date.toString();
          })
        );
        console.log("Data fetched.");
        setLoading(false);
      })
      .catch((error) => {
        console.log(error);
      });
  }, [dataType]);

  return (
    <div>
      <h4>Data fetching</h4>
      <input type="text" value={id} onChange={(e) => setId(e.target.value)} />
      <div className="date-picker">
        <label>From:</label>
        <input
          type="date"
          value={fromDate}
          onChange={(e) => setFromDate(e.target.value)}
        />
        <input
          type="time"
          value={fromTime}
          onChange={(e) => setFromTime(e.target.value)}
        />
        <label>Until:</label>
        <input
          type="date"
          value={untilDate}
          onChange={(e) => setUntilDate(e.target.value)}
        />
        <input
          type="time"
          value={untilTime}
          onChange={(e) => setUntilTime(e.target.value)}
        />
      </div>
      <button type="button" onClick={handleClick}>
        Fetch data
      </button>
      {loading && <div>Loading ...</div>}
      {!loading && (
        <div className="measurementsdata">
          <div className="measurements">
            {measurements.map((measurement) => (
              <div className="measurement">Measurement: {measurement}</div>
            ))}
          </div>
          <div className="timestamps">
            {timestamps.map((timestamp) => (
              <div className="timestamp">Timestamp: {timestamp}</div>
            ))}
          </div>
        </div>
      )}

      {/*dataMeasurements.map((measurement) => (
        <div>{measurement.data}</div>
      ))*/}
    </div>
  );
}

export default DataFetching;
