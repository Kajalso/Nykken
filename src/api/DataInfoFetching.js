import React, { useState, useEffect } from "react";
import axios from "axios";

function DataInfoFetching() {
  const [dataInfo, setDataInfo] = useState({});
  const [id, setId] = useState(1);
  const [idFromButtonClick, setIdFromButtonClick] = useState(1);

  const handleClick = () => {
    setIdFromButtonClick(id);
  };

  useEffect(() => {
    axios
      .get(`http://ibmrisvol.ibm.ntnu.no/data/info?id=${idFromButtonClick}`)
      .then((response) => {
        console.log(response.data);
        setDataInfo(response.data);
      })
      .catch((error) => {
        console.log(error);
      });
  }, [idFromButtonClick]);

  return (
    <div>
      <h4>Data info fetching</h4>
      <input type="text" value={id} onChange={(e) => setId(e.target.value)} />
      <button type="button" onClick={handleClick}>
        Fetch data info
      </button>
      <div>Sensor ID: {dataInfo.sensor_id}</div>
      <div>Description: {dataInfo.description}</div>
      <div>Unit: {dataInfo.unit}</div>
    </div>
  );
}

export default DataInfoFetching;
