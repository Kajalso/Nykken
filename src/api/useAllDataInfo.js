import { useState, useEffect } from "react";
import axios from "axios";

export const useAllDataInfo = () => {
  const [allDataInfo, setAllDataInfo] = useState([]);
  const [id, setId] = useState(1);

  useEffect(() => {
    axios
      .get(`http://ibmrisvol.ibm.ntnu.no/data/info?id=${id}`)
      .then((response) => {
        if (id <= 12) {
          setAllDataInfo((allDataInfo) => [...allDataInfo, response.data]);
          setId(id + 1);
        } else if (id === 13) {
          setAllDataInfo((allDataInfo) => [...allDataInfo, response.data]);
          console.log("All data info successfully retrieved.");
        }
      })
      .catch((error) => {
        console.log(error);
      });
  }, [id]);

  return allDataInfo;
};
