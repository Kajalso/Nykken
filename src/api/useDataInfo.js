import { useState, useEffect } from "react";
import axios from "axios";

export const useDataInfo = (id) => {
  const [dataInfo, setDataInfo] = useState([]);

  useEffect(() => {
    axios
      .get(`http://ibmrisvol.ibm.ntnu.no/data/info?id=${id}`)
      .then((response) => {
        console.log(response.data);
        setDataInfo(response.data);
      })
      .catch((error) => {
        console.log(error);
      });
  }, [id]);

  return [dataInfo, setDataInfo];
};
