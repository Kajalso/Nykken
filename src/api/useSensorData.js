import { useState, useEffect } from "react";
import axios from "axios";

export const useSensorData = (id, startDateTime, endDateTime) => {
  const [sensorData, setSensorData] = useState({
    loading: true,
    measurements: [],
  });

  const startDate = startDateTime.slice(0, 10);
  const startTime = startDateTime.slice(10, -1);

  const endDate = endDateTime.slice(0, 10);
  const endTime = endDateTime.slice(10, -1);

  useEffect(() => {
    console.log("Data with ID " + id + " is fetching...");
    setSensorData({
      loading: true,
      measurements: [],
    });
    axios
      .get(
        `http://ibmrisvol.ibm.ntnu.no/data?from=${startDate}T${startTime.slice(
          0,
          2
        )}%3A${startTime.slice(3, 5)}%3A${startTime.slice(
          6,
          8
        )}Z&until=${endDate}T${endTime.slice(0, 2)}%3A${endTime.slice(
          3,
          5
        )}%3A${endTime.slice(6, 8)}Z&identifier=${id}`
      )
      .then((response) => {
        console.log("Resonse:");
        console.log(response.data.data[0].measurements);
        let measurements = response.data.data[0].measurements;
        setSensorData({
          loading: false,
          measurements: measurements.map((data) => Number(data.measurement)),
          timestamps: measurements.map((data) => {
            let date = new Date(data.time_stamp_utc);
            return date.toString();
          }),
        });
        console.log("Data fetched.");
      })
      .catch((error) => {
        console.log(error);
        setSensorData({
          loading: false,
          measurements: [],
        });
      });
  }, [startDate, startTime, endDate, endTime, id]);

  return [sensorData, setSensorData];
};
