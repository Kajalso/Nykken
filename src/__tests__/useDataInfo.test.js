import { renderHook } from '@testing-library/react-hooks'
import axios from 'axios';

import { useDataInfo } from '../api/useDataInfo';

jest.mock('axios');

describe('rendering', () => {
    test('renders correctly', () => {
        renderHook(() => useDataInfo(1));
    });

    test('takes snapshot', () => {
        const { component } = renderHook(() => useDataInfo(1));
        expect(component).toMatchSnapshot();
    })
})

describe("testing axios", () => {
    axios.get.mockResolvedValue({
        dataInfo : {
        data_identifier: 1, 
        sensor_id: 1 ,
        unit: "\u00b0C",
        description: "Temperature",
        }
  });
  test('axios is being called', () => {
        renderHook(() => useDataInfo(1));
        expect(axios.get).toHaveBeenCalledWith(
            `http://ibmrisvol.ibm.ntnu.no/data/info?id=1`,
        );
    });
});

//Failing test
 /*    test("returns correct object", async () => {
        const { result, rerender } = renderHook(() => useDataInfo(1));
        act(()=> {
            result.current.dataInfo;
        }) 
        console.log(result.current)
        expect(result.current.dataInfo).toEqual({"data_identifier": 1, "sensor_id": 1, "unit": "\u00b0C", "description": "Temperature"})
        });
    }); */

