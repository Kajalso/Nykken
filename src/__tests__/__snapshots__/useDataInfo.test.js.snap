// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering takes snapshot 1`] = `undefined`;
